lib/Arch/X86/Runtime/BasicBlock.cpp

```diff
@@ -44,6 +48,12 @@ extern CR3Reg CR3;
 extern CR4Reg CR4;
 extern CR8Reg CR8;
 
+extern State __remill_state;
+
 #pragma clang diagnostic pop
 
+[[gnu::used]] void __remill_state_used(void) {
```

Comment:
This needs to get added for each arch (in addition to the register window struct where relevant).

Reply:
Ooooh sorry it would be easier to modify `Intrinsics.cpp` and `Instrinsics.h`!! Then you get it done for all of the architectures at once. I would also update the `IntrinsicTable` class to go and be the thing responsible for getting the state structure and register window structure (or a nullptr).

Reply:
I was thinking of that but I don't think we could put `__remill_register_window` in there since not all the runtimes have that struct.

Reply:
I guess I'll put `__remill_state` in `Intrinsics.cpp` once for all architectures. Then the `__remill_register_window` can go in each arch's `BasicBlock.cpp` where applicable.

Reply:
Oh actually, I can check `REMILL_ARCH` and only add `RegisterWindow` when it's a SPARC arch.

Reply:
[385150a](https://github.com/lifting-bits/remill/pull/596/commits/385150a857ae069698a62a683f1a0579912cf8fb)